lib/go/thrift/binary_protocol.go

```diff
@@ -26,6 +26,8 @@ import (
 	"fmt"
 	"io"
 	"math"
+
+	"github.com/google/uuid"
```

Comment:
I don't think we want to use a third party library for this:

1. so far we don't have any third party dependency (the only one is test only, not used by the library itself)
2. there are several "competing" uuid libraries and we don't really want to force our user to a particular one.

 just `type UUID [16]byte` and it would be easy enough to be casted to the majority of the third party uuid libraries.

the only thing we need to implement is converting from and to string.